File: src/SamuraiServer/SamuraiServer.Data/Models/Unit.cs
using System;

namespace SamuraiServer.Data
{
    public abstract class Unit
    {
        protected Unit()
        {
            this.CurrentHitPoints = this.HitPoints;
        }

        public Guid Id { get; set; }
        public abstract string Name { get; }
               
        public abstract string ImageSpriteResource { get; }

        public int Moves { get; set; }
        public int Size { get; set; }
        public double Attack { get; set; }
        public double Defence { get; set; }
        public double Range { get; set; }

        public double HitPoints { get; set; }
               
        public double CurrentHitPoints { get; set; }
               
        public int X { get; set; }
        public int Y { get; set; }
    }
}

File: src/Samurai.Client.JS/models/unit.js
function Unit(opts) {
  this.initialize(opts);
  this.currentHitPoints = this.hitPoints;
}

Unit.prototype.initialize = function (opt) {
  var key = null,
      default_args = {
        //Guid Id
        'Id' : null,
        'Name' : '',
        'ImageSpriteResource' : '',

        'Moves' : 0,
        //double
        'Attack' : 0,
        'Defence' : 0,
        'Range' : 0,

        'HitPoints' : 0,

        'X' : 0,
        'Y' : 0
      };

  opt || (opt = default_args);

  for(key in default_args) {
    if(typeof opt[key] == "undefined") opt[key] = default_args[key];
  }

  // opt[] has all the data - user provided and optional.
  for(key in opt) {
    this[key] = opt[key];
  }
};